src/components/alert/alert.tsx

```diff
@@ -301,6 +300,10 @@ export class Alert implements OpenCloseComponent {
 
   transitionEl: HTMLDivElement;
 
+  private transitionStartHandler = transitionStartHandler.bind(this);
+
+  private transitionEnd = transitionEnd.bind(this);
```

Comment:
Recapping our offline conversation here: I tried to implement the arrow function and pass in component but arrow function doesn't take this as an argument and event listener signature expects this as the first argument and event as second

Reply:
Recapping our online conversation here: 
I tried to implement the arrow function and pass in the component but the arrow function doesn't take this as an argument and the event listener signature expects this as the first argument and event as the second. 
The approach here was to resolve the memory leak that would happen if I did bind when passing these functions as a callback to event listeners.
Because you brought up that point, he's suggesting I implement the connect pattern instead to make it consistent with other utils.